probe_ble_specification.rst

```diff
@@ -0,0 +1,290 @@
+
+The section describes the protocol that will be sent and received over the
+Nordic UART Service.
+
+Command Header
+--------------
+
+Each message will begin with the same 5 byte header, followed by the message
+payload. The payload of each message type is described below.
+
+============== ======== ===== ===================================================================
+Value          Format   Bytes Description
+============== ======== ===== ===================================================================
+Sync Bytes     uint8_t  2     ``{ 0xCA, 0xFE }``
+Checksum       uint16_t 2     Checksum for message type, payload length + payload (sum of bytes).
+Message type   uint8_t  1
+Payload length uint8_t  1     Length of the message payload in bytes.
+============== ======== ===== ===================================================================
+
+Response Header
+---------------
+
+Each response message will include a 7 byte header with the following format.
+
+============== ======== ===== ===================================================================
+Value          Format   Bytes Description
+============== ======== ===== ===================================================================
+Sync Bytes     uint8_t  2     ``{ 0xCA, 0xFE }``
+Checksum       uint16_t 2     Checksum for message type, payload length + payload (sum of bytes).
+Message type   uint8_t  1
+Success        uint8_t  1     1 for success, 0 for failure
+Payload length uint8_t  1     Length of the message payload in bytes.
+============== ======== ===== ===================================================================
+
+Messages
+--------
+
+Request Logs (``0x04``)
+***********************
+
+The Request Logs message has the following payload. The Predictive Probe
+responds with a sequence of Log Contents messages.
+
+Request Payload
+~~~~~~~~~~~~~~~
+
+===================== ======== ===== =======================
+Value                 Format   Bytes Description
+===================== ======== ===== =======================
+Start Sequence number uint32_t 4     The first log requested
+End Sequence number   uint32_t 4     The last log requested
+===================== ======== ===== =======================
+
+Response Payload
+~~~~~~~~~~~~~~~~
+
+==================== ======== ===== ==============================
+Value                Format   Bytes Description
+==================== ======== ===== ==============================
+Sequence number      uint32_t 4     Sequence number of the record.
+Raw Temperature Data uint8_t  13    See `raw temperature data`_.
+==================== ======== ===== ==============================
+
+Common Data Formats
+###################
+
+This document defines several data formats that are common between advertising
+data and characteristic data.
+
+Product Type
+------------
+
+The product type is an enumerated value in an 8-bit (1-byte) field:
+
+* ``0``: Unknown
+* ``1``: Predictive Probe
+* ``2``: Kitchen Timer
```

Comment:
@jjohnstz -- is this supposed to be "node" for a generic MeatNet node, or Kitchen Timer?  I though I saw "node" at some place in the code?  Could be wrong.